storagemarket/provider.go

```diff
@@ -382,33 +382,48 @@ func (p *Provider) Start() ([]*dealHandler, error) {
 	}
 	log.Infow("db initialized")
 
+	// cleanup all completed deals in case Boost resumed before they were cleanedup
+	finished, err := p.dealsDB.ListCompleted(p.ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list completed deals: %w", err)
+	}
+	for i := range finished {
+		p.cleanupDealOnRestart(finished[i])
+	}
+	log.Info("finished cleaning up completed deals")
+
 	// restart all active deals
 	pds, err := p.dealsDB.ListActive(p.ctx)
 	if err != nil {
 		return nil, fmt.Errorf("failed to list active deals: %w", err)
 	}
 
-	var dhs []*dealHandler
+	// cleanup all deals that have finished successfully
+	for i := range pds {
+		deal := pds[i]
```

Comment:
```suggestion
	for _, deal := range pds {
```

Reply:
I think we need the former because deal is a pointer and we shouldn't be looping using a pointer variable, no ?

Reply:
Why not? These two are literally the same, no?

Reply:
@nonsense Look at the first answer here:
https://stackoverflow.com/questions/48826460/using-pointers-in-a-for-loop

I myself have run into this problem before and therefor stay away from using pointer vars in for loops .